docs/design/generics/details.md

```diff
@@ -2463,549 +2473,1280 @@ interface A(T:! B where ...) {
+
+-   [Equal types with different interfaces](/proposals/p2173.md#equal-types-with-different-interfaces):
+    When an associated facet is constrained to be a concrete type, it is
+    desirable for the associated facet to behave like that concrete type.
+-   [Type canonicalization](/proposals/p2173.md#type-canonicalization): to
+    enable efficient type equality.
+-   [Transitivity of equality of types](/proposals/p2173.md#transitivity-of-equality)
+
+##### Combining constraints with `&`
+
+Suppose we have `X = C where .R = A` and `Y = C where .R = B`. What should
+`C & X` produce? What should `X & Y` produce?
+
+-   Combining two rewrite rules with different rewrite targets results in a
+    facet type where the associated constant is ambiguous. Given `T:! X & Y`,
+    the type expression `T.R` is ambiguous between a rewrite to `A` and a
+    rewrite to `B`. But given `T:! X & X`, `T.R` is unambiguously rewritten to
+    `A`.
+-   Combining a constraint with a rewrite rule with a constraint with no rewrite
+    rule preserves the rewrite rule, so `C & X` is the same as `X`. For example,
+    supposing that `interface Container` extends `interface Iterable`, and
+    `Iterable` has an associated constant `Element`, the constraint
+    `Container & (Iterable where .Element = i32)` is the same as the constraint
+    `(Container & Iterable) where .Element = i32` which is the same as the
+    constraint `Container where .Element = i32`.
+
+If the rewrite for an associated constant is ambiguous, the facet type is
+rejected during [constraint resolution](#rewrite-constraint-resolution).
+
+> **Alternative considered:** We could perhaps say that `X & Y` results in a
+> facet type where the type of `R` has the union of the interface of `A` and the
+> interface of `B`, and that `C & X` similarly results in a facet type where the
+> type of `R` has the union of the interface of `A` and the interface originally
+> specified by `C`.
+
+##### Combining constraints with `and`
+
+It's possible for one `=` constraint in a `where` to refer to another. When this
+happens, the facet type `C where A and B` is interpreted as
+`(C where A) where B`, so rewrites in `A` are applied immediately to names in
+`B`, but rewrites in `B` are not applied to names in `A` until the facet type is
+[resolved](#rewrite-constraint-resolution):
+
+```carbon
+interface C {
+  let T:! type;
+  let U:! type;
+  let V:! type;
+}
+class M {
+  alias Me = Self;
+}
+// ✅ Same as `C where .T = M and .U = M.Me`, which is
+// the same as `C where .T = M and .U = M`.
+fn F[A:! C where .T = M and .U = .T.Me]() {}
+// ❌ No member `Me` in `A.T:! type`.
+fn F[A:! C where .U = .T.Me and .T = M]() {}
 ```
 
-we can constrain them to be equal in a function signature:
+##### Combining constraints with `extends`
 
-```
-fn F[MatchedPairType:! PairInterface where .Left == .Right]
-    (x: MatchedPairType*);
-```
+Within an interface or named constraint, `extends` can be used to extend a
```

Comment:
Is this still `extends` or is it now `extend`?

Reply:
Fixed. Looks like there was some other old syntax used, like `impl as` instead of `require Self impls`.